sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -950,21 +950,31 @@ CURLcode CurlSession::ReadStatusLineAndHeadersFromRawResponse(
     // The value can also be "keep-alive, close" (i.e. "both are fine"), in which case we are
     // preferring to treat it as keep-alive.
     // (https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Connection)
```

Comment:
This comment section is not so much about what is happening in the code below, but it does describe what the specs say, so I see no reason to drop the portion about `HTTP/2` and `/3`.
```suggestion
    // (https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Connection)
    // Should it come to HTTP/2 and HTTP/3, they are "keep-alive", but any response from HTTP/2 or
    // /3 containing a "Connection" header should be considered malformed. (HTTP/2:
    // https://httpwg.org/specs/rfc9113.html#ConnectionSpecific
    //  HTTP/3: https://httpwg.org/specs/rfc9114.html#rfc.section.4.2)
    //
    // HTTP/2+ are supposed to create persistent ("keep-alive") connections per host,
    // and close them by inactivity timeout. Given that we don't have such mechanism implemented
    // at this moment, we are closing all non-1.x connections immediately, which, in the worst case,
    // would only mean there's a perf hit, but the communication flow is expected to be correct.
```